### PR TITLE
Add more root files for lua-ls

### PR DESCRIPTION
**Pull request by DarkWiiPlayer**
* Busted
* LDoc
* Git

### CODE REVIEW

File: lua/lspconfig/server_configurations/lua_ls.lua [hunk 1]
```diff
@@ -1,13 +1,16 @@
 local util = require 'lspconfig.util'
 
 local root_files = {
+  '.busted',
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/lua_ls.lua:4]: `.busted` just for test case ?

File: lua/lspconfig/server_configurations/lua_ls.lua [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+  '.git',
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/lua_ls.lua:5]: why add `.git` as a file in there ? usually it's a folder?

Comment (glepnir) [reply lua/lspconfig/server_configurations/lua_ls.lua:5]: this is duplicate we have `util.find_git_ancestor(fname)` .

File: lua/lspconfig/server_configurations/lua_ls.lua [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+  '.luacheckrc',
   '.luarc.json',
   '.luarc.jsonc',
-  '.luacheckrc',
   '.stylua.toml',
-  'stylua.toml',
+  'config.ld', -- ldoc
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/lua_ls.lua:10]: this not make sense .

Comment (DarkWiiPlayer) [reply lua/lspconfig/server_configurations/lua_ls.lua:10]: `config.ld` is the project-level config file for ldoc

**Comment by DarkWiiPlayer** [comment]
I just spotted the `find_git_ancestor` at the end of the function.

Does this make sense as the last option? I've repeatedly run into trouble because my Lua projects are inside a `lua` directory and even if they are a git repository, the function will first return the `lua` dir before even getting to the git check.

A workaround that would work both for git repositories inside a `lua` directory as well as larger multi-language projects with Lua code in a `lua` subdirectory would be to prefer the longer of the two paths, if both are found.

**Pull request closed without merging**